# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: ddb496eb39ab
# Full Hash: ddb496eb39ab8e4a48b7e6924a3a908f30620864
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 21:46:09
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add a dedicated method for resetting storages of a specific client; r=dom-storage-reviewers,jari
#   
#   Client storages can be currently reset by using the optional argument of
#   nsIQuotaManagerService::ResetStoragesForPrincipal. However, supporting
#   resetting storages for both a principal and a client is not ideal for several
# ==============================================================================

diff -r 67a39d84be12 -r ddb496eb39ab dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Wed Oct 09 11:59:34 2024 +0000
+++ b/dom/quota/ActorsParent.cpp	Wed Oct 09 11:59:34 2024 +0000
@@ -5944,6 +5944,22 @@
   return shutdownOriginOp->OnResults();
 }
 
+RefPtr<BoolPromise> QuotaManager::ShutdownStoragesForClient(
+    Maybe<PersistenceType> aPersistenceType,
+    const PrincipalInfo& aPrincipalInfo, Client::Type aClientType) {
+  AssertIsOnOwningThread();
+
+  auto shutdownClientOp =
+      CreateShutdownClientOp(WrapMovingNotNullUnchecked(this), aPersistenceType,
+                             aPrincipalInfo, aClientType);
+
+  RegisterNormalOriginOp(*shutdownClientOp);
+
+  shutdownClientOp->RunImmediately();
+
+  return shutdownClientOp->OnResults();
+}
+
 RefPtr<BoolPromise> QuotaManager::ShutdownStorage(
     Maybe<OriginOperationCallbackOptions> aCallbackOptions,
     Maybe<OriginOperationCallbacks&> aCallbacks) {